app/controllers/presence_controller.rb

```diff
@@ -5,21 +5,26 @@ class PresenceController < ApplicationController
   before_action :ensure_logged_in, only: [:update]
 
   def get
-    name = params.require(:channel)
+    names = params.require(:channels)
+    raise Discourse::InvalidParameters.new(:channels) if !(names.is_a?(Array) && names.all? { |n| n.is_a? String })
 
-    begin
+    names.uniq!
+
+    raise Discourse::InvalidParameters.new("Too many channels") if names.length > 50
+
+    result = {}
+    names.each do |name|
       channel = PresenceChannel.new(name)
+      if channel.can_view?(user_id: current_user&.id)
```

Comment:
I have a small concern here, `channel.can_view?` triggers a DB query if the channel is configured with `allowed_group_ids`. The worst case here is that this endpoint will trigger 50 DB query which I think is not ideal. Perhaps we need a way to bulk check `can_view?` for a bunch of channels against a single user?